das/redis_storage_service_test.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2021-2022, Offchain Labs, Inc.
+// For license information, see https://github.com/nitro/blob/master/LICENSE
+
+package das
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/alicebob/miniredis/v2"
+	
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+func TestRedisStorageService(t *testing.T) {
+	ctx := context.Background()
+	timeout := uint64(time.Now().Add(time.Hour).Unix())
+	baseStorageService := NewMemoryBackedStorageService(ctx)
+	server, err := miniredis.Run()
+	Require(t, err)
+	redisService, err := NewRedisStorageService(
+		RedisConfig{
+			"redis://" + server.Addr(),
+			time.Hour,
+			"b561f5d5d98debc783aa8a1472d67ec3bcd532a1c8d95e5cb23caa70c649f7c9",
+		}, baseStorageService)
+
+	Require(t, err)
+
+	val1 := []byte("The first value")
+	val1CorrectKey := crypto.Keccak256(val1)
+	val1IncorrectKey := crypto.Keccak256(append(val1, 0))
+
+	_, err = redisService.GetByHash(ctx, val1CorrectKey)
+	if !errors.Is(err, ErrNotFound) {
+		t.Fatal(err)
+	}
+
+	err = redisService.Put(ctx, val1, timeout)
+	Require(t, err)
+
+	_, err = redisService.GetByHash(ctx, val1IncorrectKey)
+	if !errors.Is(err, ErrNotFound) {
+		t.Fatal(err)
+	}
+	val, err := redisService.GetByHash(ctx, val1CorrectKey)
+	Require(t, err)
+	if !bytes.Equal(val, val1) {
+		t.Fatal(val, val1)
+	}
+
+	//For Case where the value is present in the base storage but not present in the cache.
```

Comment:
As above, would be good to test that the expected data is written into, and retrievable from, the Redis layer.